### PR TITLE
add error handling

### PR DESCRIPTION
**Pull request by braxex**
## Description

Adds some basic error handling so we can catch those errors and show 'em to the people.

### How to test

1. Pull and run.
2. Try to log in with invalid credentials. You should get a browser alert with an error message.

### Changed Dependencies

No changed deps.

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research) tests
- [x] ~~I have added analytics, if necessary~~
- [x] ~~I have updated the documentation, if necessary~~

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/utils/auth/useUser.ts [hunk 1]
```diff
@@ -4,10 +4,11 @@ import { UserContext } from "./userContext";
 export const useUser = (): any => {
   const context = useContext(UserContext);
 
-  if (context === undefined) {
+  if (typeof context === "undefined") {
```
Comment (bodnarbm) [inline services/ui-src/src/utils/auth/useUser.ts:7]: Why this change? Just wondering for my own learning. I have a preference for direct checks against undefined or loose equality check against `null` (e.g., `null == context`). But just wondering what this typeof covers over the string equality with `undefined`?

**Review by gmrabian — APPROVED** [review]

**Review by bodnarbm — APPROVED** [review]

File: services/ui-src/src/utils/auth/useUser.ts (hunk 1 above)
Comment (braxex) [reply services/ui-src/src/utils/auth/useUser.ts:7]: They're very similar, but using `typeof` it won't throw an error if the variable hasn't been declared, so I always default to using `typeof` and strict equality when checking for undefined. It's an especially common pattern in SSR apps to guard server-side vs. client-side code by checking for the window obj, like:

```
if (typeof window !== 'undefined') {
  // do whatever
}
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#typeof_operator_and_undefined
